### PR TITLE
[Refactoring] Use c++11 list initialization for maps/vectors

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Nukes the dependency on Boost's `list_of.hpp`

Also introduce UniValueType

UniValueType is a wrapper for UniValue::VType which allows setting
a typeAny flag. This flag indicates the type does not matter.

### CODE REVIEW

**Comment by Fuzzbawls** [comment]
rebased

File: src/chainparams.cpp [hunk 1]
```diff
@@ -64,31 +62,31 @@ static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits
...
+    { 908000, uint256S("202708f8c289b676fceb832a079ff6b308a28608339acbf7584de533619d014d")},
+    {1142400, uint256S("98aff9d605bf123247f98b1e3a02567eb5799d208d78ec30fb89737b1c1f79c5")},
+    {1679090, uint256S("f747ce055ba1b12e1f2e842bd480bc647210799359cb2e553ab292065e3419d6")}, //!< First block with a "wrapped" serial spend
+    {1686229, uint256S("bb42bf1e886a7c23474634c90893dd3d68a6ccbfea4ac92a98da5cad0c6a6cb7")}, //!< Last block in the "wrapped" serial attack range
+    {1778954, uint256S("0d3241268264a2908d6babf00d9cd1ffb83d93d7bb4e428820127fe227c2029c")}, //!< Network split here
+    {1788528, uint256S("ea9243ff8fc079fdd7a04f11fac415de4d98e1bb0dc38db6f79f8f8bbfdbe496")}, //!< Network split here
+    {2153200, uint256S("14e477e597d24549cac5e59d97d32155e6ec2861c1003b42d0566f9bf39b65d5")}, //!< First v7 block
+    {2356049, uint256S("62e80d8e193bca84655fb78893b20f54a79f2d71124c4ea37b7ef51a0d5451c4")}, //!< Network split here
+    {2365700, uint256S("b5d0beead57735539abc2db2b0b08cd65db3e5928efd3c3bf3182d5bf013f36c")}, //!< PIVX v4.1.1 enforced
+    {2678402, uint256S("580a26ff0a45177a7a6f387f009c5b26140ea48b4790a857d9a796f8b3c25899")}, //!< Network split here}
```
Comment (random-zebra) [inline src/chainparams.cpp:88]: nit: extra `}` at the end of the comment.

**Review by random-zebra — APPROVED** [review]
ACK 72755979076f0a08d6c469363bebbd1e55046c86

**Review by furszy — APPROVED** [review]
utACK 7275597 and merging..

will tackle zebra's comment in my next PR that updates all of this area, no need to wait for it here.

**Pull request merged**